From 2a1522ae5240824c1069655c203ddde4c331035c Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 16 Aug 2017 18:28:15 +0800
Subject: [PATCH 2386/5242] MLK-16197-11 video: fbdev: dcss: remove (un)blank
 calls in 'dcss_set_par'

commit  2d54c268ebd0644bcf8b2544a2f855b822fc75aa from
https://source.codeaurora.org/external/imx/linux-imx.git

Remove the possible 'dcss_blank' calls in 'dcss_set_par',
since the context loader can change the DCSS sub-modules
configuration on the fly.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |   16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 8594d5c..f8baa6c 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2661,7 +2661,7 @@ static int config_channel_pipe(struct dcss_channel_info *cinfo)
 
 static int dcss_set_par(struct fb_info *fbi)
 {
-	int ret = 0, saved_blank;
+	int ret = 0;
 	int fb_node = fbi->node;
 	struct dcss_channel_info *cinfo = fbi->par;
 	struct dcss_info *info = cinfo->dev_data;
@@ -2676,15 +2676,6 @@ static int dcss_set_par(struct fb_info *fbi)
 	chan_info = &chans->chan_info[fb_node];
 	cb = &chan_info->cb;
 
-	saved_blank = chan_info->blank;
-
-	if (saved_blank == FB_BLANK_UNBLANK) {
-		/* blank the fb if is not to
-		 * stop dcss for a while
-		 */
-		ret = dcss_blank(FB_BLANK_NORMAL, fbi);
-	}
-
 	/* TODO: add save/recovery when config failed */
 	fb_var_to_pixmap(&chan_info->input, &fbi->var);
 
@@ -2713,11 +2704,6 @@ static int dcss_set_par(struct fb_info *fbi)
 	}
 #endif
 
-	if (saved_blank == FB_BLANK_UNBLANK) {
-		/* unblank the fb if need */
-		ret = dcss_blank(FB_BLANK_UNBLANK, fbi);
-	}
-
 	goto out;
 
 fail:
-- 
1.7.9.5

